### DISCUSSION TITLE
Cannot install brew - Big Sur - fatal: Could not read from remote repository.

### DISCUSSION THREAD

**Discussion by guzunov1990bg** [Homebrew/discussions#1721] [Getting started]
Hi all,

It's been a while since I've set up a Mac, but I've never had any issues with installing brew?

I have BigSur 11.4, using terminal (bash) I copied the command found on the brew.sh page:

/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

and I got the following result:

==> Checking for `sudo` access (which may request your password).
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew
==> HOMEBREW_BREW_GIT_REMOTE is set to a non-default URL:
... will be used for Homebrew/brew Git remote.
==> HOMEBREW_CORE_GIT_REMOTE is set to a non-default URL:
... will be used for Homebrew/core Git remote.

Press RETURN to continue or any other key to abort
==> /usr/bin/sudo /bin/mkdir -p /usr/local/Homebrew
==> /usr/bin/sudo /usr/sbin/chown -R guzunov:admin /usr/local/Homebrew
==> Downloading and installing Homebrew...
fatal: '...' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Failed during: git fetch --force origin

Is there any issue with the github repo? I've tried to copy the script locally and run it, but I have the same error?

Any suggestions?

**Comment by carlocab** [comment]
Try
```
unset HOMEBREW_BREW_GIT_REMOTE HOMEBREW_CORE_GIT_REMOTE
```
then run the installer again.

**Reply by guzunov1990bg** [reply]
@carlocab that definitely worked, did I miss a step somewhere or that's needed as a prerequisite before installing?

**Reply by carlocab** [reply]
Not really. You're free to set them -- as long as they point to valid Homebrew repositories or mirrors thereof. It looks like yours referred to broken ones. That's why you see this message:
```
==> HOMEBREW_BREW_GIT_REMOTE is set to a non-default URL:
... will be used for Homebrew/brew Git remote.
==> HOMEBREW_CORE_GIT_REMOTE is set to a non-default URL:
... will be used for Homebrew/core Git remote.
```

**Reply by guzunov1990bg** [reply]
Got it, thanks for the prompt reply, I hope this will help some other lost souls :)

**Reply by drubio5-bdb** [reply]
Thank u so much

**Reply by bryan-melo** [reply]
You saved me from an endless rabit hole! Thank you!

**Reply by fhussein24** [reply]
Thank you so much. This really helped

**Reply by NoamFav** [reply]
thank you so much, i tried for hours with no success, you saved my sleep schedule XD

**Reply by mr008** [reply]
you saved me hours!!

**Reply by waleoyediran** [reply]
Its 2023 and you are still saving lives :)

**Reply by AdnanSharifov** [reply]
Thank you sooooo much m8

**Reply by naizapp** [reply]
Thank you so much

**Reply by mt467** [reply]
Magic!

**Reply by ducnhat1309** [reply]
you  saved me!

**Reply by Neyan** [reply]
Thank you !

**Reply by HaiLe81** [reply]
Thank you!

**Reply by Caraluvian** [reply]
Thank you!!!

**Reply by JerMa88** [reply]
Thank you! It worked like a charm on both of my M-series mac!

**Reply by divikshrivastava** [reply]
Thanks! saved me works awesome on my Mac

**Comment by ubourdon** [comment]
Hi. I have exactly the same error when try to install brew on Big Sur.
But the tips `unset HOMEBREW_BREW_GIT_REMOTE HOMEBREW_CORE_GIT_REMOTE` don't work for me.

**Reply by ubourdon** [reply]
The only error message i have is `Failed during: git fetch --force origin`. Nothng else

**Reply by ubourdon** [reply]
Ok the error was ulumit -f config value that bug git when clone repo with too large file

**Comment by takujoye14** [comment]
when I run this:   /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" to install brew, I get this error: fatal: unable to connect to github.com:
github.com[0: 140.82.121.4]: errno=Operation timed out

Failed during: /usr/bin/git fetch --force origin
Does anyone know how to fix it?
